File: MarketplaceJPA/src/main/java/com/skilldistillery/marketplace/entities/Bid.java
 package com.skilldistillery.marketplace.entities;

 import org.hibernate.annotations.CreationTimestamp;

 import javax.persistence.*;
 import java.time.LocalDateTime;

@Entity
public class Bid {
	@Id
	@GeneratedValue(strategy= GenerationType.IDENTITY)
	private int id;

	@ManyToOne
	@JoinColumn(name="token_id")
	private Token token;
	
	private String description;
	
	@OneToOne
	@JoinColumn(name="seller_id")
	private User seller;
	
	@OneToOne
	@JoinColumn(name="buyer_id")
	private User buyer;
	
	@Column(name="bid_date")
	@CreationTimestamp
	private LocalDateTime bidDate;
	
	@Column(name="offer_amount")
	private double offerAmount;
	
	public double getOfferAmount() {
		return offerAmount;
	}

	public void setOfferAmount(double offerAmount) {
		this.offerAmount = offerAmount;
	}

	public void setBidDate(LocalDateTime bidDate) {
		this.bidDate = bidDate;
	}

	
	public User getSeller() {
		return seller;
	}

	public void setSeller(User seller) {
		this.seller = seller;
	}

	public User getBuyer() {
		return buyer;
	}

	public void setBuyer(User buyer) {
		this.buyer = buyer;
	}

	
	public Bid() { super(); }

	public Bid(int id, Token token, String description, LocalDateTime transferDate, User buyer, User seller) {
		
		this.token = token;
		this.description = description;
		this.bidDate = transferDate;
		this.seller=seller;
		this.buyer=buyer;
		
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public Token getToken() {
		return token;
	}

	public void setToken(Token token) {
		this.token = token;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public LocalDateTime getBidDate() {
		return bidDate;
	}

	public void getBidDate(LocalDateTime bidDate) {
		this.bidDate = bidDate;
	}

	@Override
	public String toString() {
		return "Bid [id=" + id + ", token=" + token + ", description=" + description + ", seller=" + seller + ", buyer="
				+ buyer + ", bidDate=" + bidDate + ", offerAmount=" + offerAmount + "]";
	}
	
	
}

File: MarketplaceJPA/src/main/java/com/skilldistillery/marketplace/entities/Bid.kt
package com.skilldistillery.marketplace.entities

import org.hibernate.annotations.CreationTimestamp
import java.math.BigDecimal
import java.time.LocalDateTime
import javax.persistence.*

@Entity
data class Bid (
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int = 0,
    @ManyToOne @JoinColumn(name = "token_id")
    var token: Token? = null,
    var description: String? = null,
    @OneToOne @JoinColumn(name = "seller_id")
    var seller: User? = null,
    @OneToOne @JoinColumn(name = "buyer_id")
    var buyer: User? = null,
    @Column(name = "bid_date") @CreationTimestamp
    var bidDate: LocalDateTime? = null,
    @Column(name = "offer_amount", precision = 20, scale = 8)
    var offerAmount: BigDecimal,
    @ManyToOne(fetch = FetchType.LAZY)
    val auction: Auction? = null
    ) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Bid

        return id == other.id
    }

    override fun hashCode(): Int {
        return id
    }

    override fun toString(): String {
        return "Bid(id=$id, token=$token, description=$description, seller=$seller, buyer=$buyer, bidDate=$bidDate, offerAmount=$offerAmount)"
    }
}
